encoding/int/encoding.go

```diff
@@ -0,0 +1,42 @@
+package int
+
+import "io"
+
+// Encoder encodes int values.
+type Encoder interface {
+	// Encode a block if ints.
+	Encode(values []int) error
+	// Bytes reutrns the encoded bytes of the encoder.
+	// The format of the byte slice is as follows:
+	//   - version:                 1 byte
+	//   - metadata length bytes:   VInt
+	//   - metadata bytes:          Variable bytes
+	//   - data bytes:              Variable bytes
+	Bytes() []byte
+	// Reset internal memory buffers manually.
+	// Otherwise, buffers are reset upon the next call to `Encode`.
+	Reset()
```

Comment:
I'd perfer to document that the callers should always reset buffers before calling `Encode`, instead of implicitly relying on the `Encode` API to reset the buffers for the sake of the "least responsibilities` principle.

Reply:
Can you remove the `Otherwise, ...` comment if that no longer applies?